ui/src/app/pages/admin/admin-workspace-search.tsx

```diff
@@ -0,0 +1,62 @@
+import {Component} from '@angular/core';
+import {Button} from 'app/components/buttons';
+import {FlexRow} from 'app/components/flex';
+import {TextInput} from 'app/components/inputs';
+import colors from 'app/styles/colors';
+import {ReactWrapperBase, UrlParamsProps, withUrlParams} from 'app/utils';
+import {navigate} from 'app/utils/navigation';
+import * as React from 'react';
+
+interface State {
+  workspaceNamespace: string;
+}
+
+class AdminWorkspaceSearchImpl extends React.Component<UrlParamsProps, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      workspaceNamespace: '',
+    };
+  }
+
+  navigateToWorkspace() {
+    navigate(['/admin/workspaces/' + this.state.workspaceNamespace]);
+  }
+
+  render() {
+    return <FlexRow style={{justifyContent: 'flex-start', alignItems: 'center', marginTop: '1rem'}}>
+        <label style={{color: colors.primary, marginRight: '1rem'}}>Workspace namespace</label>
+        <TextInput
+            style={{
+              width: '10rem',
+              marginRight: '1rem'
+            }}
+            onChange={value => this.setState({workspaceNamespace: value})}
+            onKeyDown={(event: KeyboardEvent) => {
```

Comment:
ty for keeping the enter-to-search though.

Reply:
I would die if this went away. Other things that I am unreasonably bothered by: not being able to ctrl-click links to open in a new tab (basically all of our in-app links don't support this), and not being able to cmd-enter to perform primary action on modal dialogs. It may not happen soon, but eventually I will fix all these things :)